### DISCUSSION TITLE
[Fooocus 2.2.1] Bugfixes, Array Syntax Prompt Weight

### DISCUSSION THREAD

**Discussion by mashb1t** [lllyasviel/Fooocus#2440] [General]
## What's Changed
* fix: adjust parameters for upscale fast 2x by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/2411
* fix: add handling for filepaths to image grid by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/2414
* fix: add fallback value for default_max_lora_number when default_loras is empty by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/2430
* feat: add metadata flag and steps override to history log by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/2425
* fix: adjust width of lora weight for firefox by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/2431
* fix: add hint for png to metadata scheme selection by @eddyizm in https://github.com/lllyasviel/Fooocus/pull/2434
* fix: typo in wildcards/animal.txt by @nbs in https://github.com/lllyasviel/Fooocus/pull/2433
* feat: match anything in array syntax, not only words and whitespace by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/2438

## New Contributors
* @nbs made their first contribution in https://github.com/lllyasviel/Fooocus/pull/2433

**Full Changelog**: https://github.com/lllyasviel/Fooocus/compare/2.2.0...2.2.1

<hr /><em>This discussion was created from the release <a href='https://github.com/lllyasviel/Fooocus/releases/tag/2.2.1'>2.2.1</a>.</em>

**Comment by ILikeToasters** [comment]
Thanks for the hard work! Excited to use prompt weights in the arrays!

**Comment by marcoscattani** [comment]
Thank you for sharing this amazing tool with everyone!

**Comment by bsdesigner** [comment]
It turned out that the "Skip" button does not work. When you press it, it interrupts not the current generation, but resets the generation altogether.....
![Снимок экрана 2024-03-07 011601](https://github.com/lllyasviel/Fooocus/assets/36730003/34a64e84-d2aa-4d11-b16b-66ea8869b749)

**Reply by mashb1t** [reply]
This is what the skip button does, it skips the current image sampling and start the next  sampling at step 0. Not certain if your expectations are different.
Feel free to create an issue if there is an unexpected behavior and it is reproducible for you.

**Reply by bsdesigner** [reply]
That's the thing, it used to skip streaming generation (let's say number 2 of 6) and start generating numbers 3, 4 and so on. But now when you press Skip, all generation stops!

**Reply by mashb1t** [reply]
Not reproducible for me on all of my four platforms, 2x Windows, Macbook and Colab.

>  Feel free to create an issue if there is an unexpected behavior and it is reproducible for you.

**Reply by bsdesigner** [reply]
https://github.com/lllyasviel/Fooocus/assets/36730003/da8ba7f5-7754-42de-b0ee-ee952be16c9f

**Reply by bsdesigner** [reply]
You fixed it that fast? It wasn't working, and now it's working!!! Now that's efficiency!!!!! Thank you so much!

**Reply by mashb1t** [reply]
uhm... yeah, was obv. all me! (jokes aside, didn't touch anything) - case closed

**Comment by paperclip52** [comment]
They really, really, really have to add an erasure brush to the inpaint outpaint! It makes no sense having to step back 2,3,4,5, times or delete the whole mask, when you could just switch to the erasure bush.....and erase. Save a whole lota time and aggravation IMO. It also would be nice if they let you manually change the dimensions in the outpaint (top, bottom, right, left} instead of having to use the default dimensions.

**Reply by mashb1t** [reply]
Gradio 4.x has an image editor, which allows for this functionality, introduced in https://github.com/gradio-app/gradio/pull/6169
Fooocus.
Fooocus uses Gradio 3.x and currently noone is working on the upgrade to 4.x.

**Comment by yorai1212** [comment]
GitHub noob here, how do I download this update? (2.2.1)

**Reply by mashb1t** [reply]
Please follow the instructions in the readme depending on your system.
For Windows e.g. it's easiest if you download the 7zip file, extract it and double-click the included run.bat file.

**Reply by yorai1212** [reply]
Yeah no, that's not my problem, not with installing.
The problem is where I can find the download file even.
![image](https://github.com/lllyasviel/Fooocus/assets/32553260/ab2b95d2-53c0-4953-b411-543f37cc8111)
As you can see ^ I can't find the download file.

**Reply by bsdesigner** [reply]
![изображение_2024-03-09_005603730](https://github.com/lllyasviel/Fooocus/assets/36730003/6783f6b1-9ebc-4a35-b730-e7f50ff74c3b)

**Reply by mashb1t** [reply]
@bsdesigner this is the source code, not the "download file" which @yorai1212 is referring to.
@yorai1212 please check the readme and search for download, you'll be surprised how obvious it is. It's hard to miss, trust me.

**Reply by bsdesigner** [reply]
Is this the link you were referring to?
![Снимок экрана 2024-03-09 025504](https://github.com/lllyasviel/Fooocus/assets/36730003/f0b3ce5a-3334-457b-9058-cf76d6eb806f)

**Reply by eddyizm** [reply]
@bsdesigner that's the one.

**Reply by yorai1212** [reply]
@mashb1t Right, I'm aware of that button. But that's version 2.1.831. Right?
Where can I find version 2.2.1?

**Reply by mashb1t** [reply]
@yorai1212 it auto-updates using git pull, no worries! The 7z file will be updated when adding new packages.

**Comment by yorai1212** [comment]
@mashb1t Oh okay, I see, thank you!
GitHub is so confusing to me sometimes.. hehe

**Reply by eddyizm** [reply]
@yorai1212 FYI - when you launch it, you can see the version as it gets printed to the terminal. Notice my folder says 2.1.831 but the version printed when launched below is 2.2.1
```
[System ARGV] ['K:/Apps/Fooocus_win64_2-1-831/Fooocus/entry_with_update.py', '--listen', '--theme', 'dark', '--preset', 'turbo']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.2.1
Loaded preset:
```

**Reply by yorai1212** [reply]
Oh okay cool, downloaded it now and yeap.. 2.2.1 for me too.
I also just realized I didn't reply but created a new comment, my bad.

I'm guessing I should wait like a year or 2 until AI becomes faster for AMD GPUs. Each generation takes a long time..
While there's reports from Nvidia GPUs users taking just a few seconds for a generation.
